### PR TITLE
hopefully fix docs

### PR DESCRIPTION
**Pull request by ExpandingMan**
This will hopefully fix the docs, not sure how I missed this line.

cc @rikhuijzer (who has been advising me on package deployment stuff, thanks btw)

### CODE REVIEW

File: docs/make.jl [hunk 1]
```diff
@@ -18,3 +18,5 @@ makedocs(;
         "API" => "api.md",
        ],
    )
+
+deploydocs(repo="github.com/dmlc/XGBoost.jl", devbranch="master")
```
Comment (rikhuijzer) [inline docs/make.jl:22]:
```suggestion
deploydocs(repo="github.com/dmlc/XGBoost.jl.git", devbranch="master")
```
Documenter.jl is very pedantic about the `.git`

**Comment by rikhuijzer** [comment]
I also now notice that this repository always sets `DOCUMENTER_KEY` **and** `GITHUB_TOKEN` which is uncommon. However, based on the successful CompatHelper-triggered CI runs, the `DOCUMENTER_KEY` is most likely set so that should work.

**Comment by aviks** [comment]
Thanks, hopefully this works. I've forgotten how this works these days, I just let Juliahub build and host docs for my packages these days, (but I'm biased 😄 )

**Comment by rikhuijzer** [comment]
I'm not sure what's going on now. Normally, GitHub Pages takes at most half an hour to update the website once the CI reports that the Pages deploy succeeded and we're now one hour after the CI. The files in `gh-pages` appear correct also.

My current guess is that the Pages have been disabled for this repository. 

Can someone with the right permissions go into the repository settings for this project and verify that the Pages are enabled?

**Comment by aviks** [comment]
I don't have access, maybe @trivialfis might be able to help?

**Comment by trivialfis** [comment]
Apologies, I don't know how the document is deployed for this repository.

**Comment by ExpandingMan** [comment]
@trivialfis it should only be a github setting to tell it to take the docs page from the `gh-pages` branch which should already present.  See the settings page pictured above.

**Comment by trivialfis** [comment]
unfortunately, I don't have access to the setting. I help some development for xgboost and its python binding (recently the R binding as well). But CI for xgboost is maintained by @hcho3 , and setting access are limited to @tqchen and @hcho3 .

**Pull request merged**